src/transformStateWithClones.js

```
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateIterations = [];
+  const stateIteration = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateIteration, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateIteration[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in stateIteration) {
+          delete stateIteration[key];
+        }
+        break;
+
+      default: break;
```

It is better to return an error or a message that something went wrong.